modules/storage/src/service/fs.rs

```diff
@@ -94,7 +93,9 @@ impl StorageBackend for FileSystemBackend {
         // set up reader
 
         let mut stream = pin!(stream);
-        let mut digest = Sha256::new();
+        let mut sha256 = Sha256::new();
+        let mut sha384 = Sha384::new();
+        let mut sha512 = Sha512::new();
```

Comment:
I don't quite understand why storage needs to calculate all these for every write. It's only gonna use one of them, correct?

Reply:
Correct. But it returns them after the insert. My idea was to refactor in another step, passing those into the loaders, so that they might be able to use that info. Right now, they do this during the read. Which also works.